packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx

```diff
@@ -0,0 +1,321 @@
+              break;
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+            break;
+        }
+      },
+      [apiRef, colDef.field, colIndex, columnFields, headerFilterMenuRef, isEditing, isMenuOpen],
+    );
+
+    const publish = React.useCallback(
+      (eventName: keyof GridHeaderFilterEventLookup, propHandler?: React.EventHandler<any>) =>
+        (event: React.SyntheticEvent) => {
+          apiRef.current.publishEvent(
+            eventName,
+            apiRef.current.getColumnHeaderParams(colDef.field),
+            event as any,
+          );
+
+          if (propHandler) {
+            propHandler(event);
+          }
+        },
+      [apiRef, colDef.field],
+    );
+
+    const onMouseDown = React.useCallback(
+      (event: React.MouseEvent) => {
+        if (!hasFocus) {
+          if (inputRef.current) {
+            inputRef.current.focus();
+          }
+          apiRef.current.setColumnHeaderFilterFocus(colDef.field, event);
+        }
+      },
+      [apiRef, colDef.field, hasFocus],
+    );
+
+    const mouseEventsHandlers = React.useMemo(
+      () => ({
+        onKeyDown: publish('headerFilterKeyDown', onKeyDown),
+        onClick: publish('headerFilterClick'),
+        onMouseDown: publish('headerFilterMouseDown', onMouseDown),
+      }),
+      [onMouseDown, onKeyDown, publish],
+    );
+
+    const ownerState = {
+      ...rootProps,
+      colDef,
+    };
+
+    const classes = useUtilityClasses(ownerState as OwnerState);
+
+    const isNoInputOperator =
+      filterOperators?.find(({ value }) => item.operator === value)?.requiresFilterValue === false;
+
+    const isApplied = Boolean(item?.value) || isNoInputOperator;
+    const label =
+      currentOperator.headerLabel ??
+      apiRef.current.getLocaleText(
+        `headerFilterOperator${capitalize(item.operator)}` as 'headerFilterOperatorContains',
+      );
+
+    const isFilterActive = isApplied || hasFocus;
+
+    return (
+      <div
+        className={clsx(classes.root, headerClassName)}
+        ref={handleRef}
+        style={{
+          height,
+          width,
+          minWidth: width,
+          maxWidth: width,
+        }}
+        role="columnheader"
+        aria-colindex={colIndex + 1}
+        aria-label={headerFilterComponent == null ? colDef.headerName ?? colDef.field : undefined}
+        {...other}
+        {...mouseEventsHandlers}
+      >
+        {headerFilterComponent}
+        {InputComponent && headerFilterComponent === undefined ? (
+          <InputComponent
+            apiRef={apiRef}
+            item={item}
+            inputRef={inputRef}
+            applyValue={applyFilterChanges}
+            onFocus={() => apiRef.current.startHeaderFilterEditMode(colDef.field)}
+            onBlur={() => apiRef.current.stopHeaderFilterEditMode()}
+            placeholder={apiRef.current.getLocaleText('columnMenuFilter')}
+            label={isFilterActive ? capitalize(label) : ' '}
```

Comment:
I think that we should change this logic, turning the label to an empty string breaks the floating label animation.

Reply:
Agreed, one reason it happens is that label also changes on focus (etc `Filter` -> `Contains`), added to the list of follow-up [items](https://github.com/mui/mui-x/issues/9034).